### PR TITLE
[mmu probing] Add ProbingBase PG counter true env UT coverage

### PR DESCRIPTION
**Pull request by yyynini**
### Description of PR
Add unit test coverage for the real `ProbingBase.setUp()` environment parsing path for `INGRESS_DROP_USE_PG_COUNTER=true`.

This follows reviewer guidance to improve mock UT coverage for existing supported behavior instead of adding edge-case tests that may force unfinished algorithm behavior.

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
- [x] Test case improvement

#### What is the motivation for this PR?
Improve coverage for the real `ProbingBase.setUp()` path. Existing tests simulated the PG-counter environment parsing logic directly, but did not execute `setUp()` itself.

#### How did you do it?
Added a mock-friendly UT that calls `ProbingBase.setUp()` and verifies `INGRESS_DROP_USE_PG_COUNTER=true` enables PG drop counter mode, while mocking PTF/switch initialization dependencies.

#### How did you verify/test it?
Mock UT/IT result:

```text
cd tests/saitests/mock/ut
python3 -m pytest -v test_probing_base.py::TestProbingBaseSetUp::test_setUp_applies_ingress_drop_pg_counter_true_env
  -> passed

cd tests/saitests/mock/ut
python3 -m pytest -v .
  -> 446 passed, 1 failed, 4 warnings

cd tests/saitests/mock/it
python3 -m pytest -v .
  -> 75 passed
```

Failure note:

The single full mock UT failure is unrelated to this PR and reproducible on the baseline branch:

```text
test_probing_base.py::TestProbingBaseGetRxPort::test_get_rx_port_wrapper
AssertionError: Should log before and after
assert 1 == 2
```

That baseline test expects two `log_message()` calls from `ProbingBase.get_rx_port()`. Current code emits one `log_message()` call and uses `ProbingObserver.trace()` for the retry/resolution path. This PR only adds `ProbingBase.setUp()` env parsing coverage and does not touch `get_rx_port()`.

### CODE REVIEW

File: tests/saitests/mock/ut/test_probing_base.py [hunk 1]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
         assert pb.use_pg_drop_counter is False
         print("[OK] use_pg_drop_counter defaults to False")
 
+    @pytest.mark.order(9372)
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py]: Minor — the order number `9372` looks like a typo. The whole file uses a strict 1-by-1 incrementing convention (910 → 940), and sister tests in `TestProbingBaseSetUp` are 933-937. `9372` causes this test to run *after* the tests in `TestProbingBaseTearDown` (938), `TestProbingBaseRunTest` (939/940), and `TestProbingBaseGetRxPort` (940) due to numeric sort.

No functional impact (pytest-order is deterministic and tests don't share state), just a maintainability nit.

Suggested:
```python
@pytest.mark.order(9375)  # or 941 to append at the end of the file's sequence
```

File: tests/saitests/mock/ut/test_probing_base.py [hunk 2]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
...
+    def test_setUp_applies_ingress_drop_pg_counter_true_env(self):
+        """Test real setUp() applies INGRESS_DROP_USE_PG_COUNTER=true"""
+        print("\n=== Testing setUp() - Real INGRESS_DROP_USE_PG_COUNTER=true Env ===")
+
+        class MockThriftInterfaceDataPlane:
+            @staticmethod
+            def setUp(_test):
+                return None
+
+        pb = ConcreteProbingBase()
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py:619]: Minor — consider adding a baseline guard before `setUp()` runs:

```python
pb = ConcreteProbingBase()
assert not hasattr(pb, 'use_pg_drop_counter'), \
    "test relies on setUp() being the sole setter of use_pg_drop_counter"
pb.clients = Mock()
```

Rationale: `ConcreteProbingBase.__init__` deliberately doesn't initialize `use_pg_drop_counter` so `setUp()` is the sole setter. But if someone later adds `self.use_pg_drop_counter = True` to `__init__` (e.g., as documentation of a default), `setUp()` could silently skip the PG-counter block and this test would still pass — silent false positive.

File: tests/saitests/mock/ut/test_probing_base.py [hunk 3]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
...
+        pb.clients = Mock()
+
+        with patch('probing_base.sai_base_test.ThriftInterfaceDataPlane',
+                   MockThriftInterfaceDataPlane):
+            with patch('probing_base.switch_init') as mock_switch_init:
+                with patch('probing_base.time.sleep') as mock_sleep:
+                    with patch('probing_observer.ProbingObserver.trace') as mock_trace:
+                        with patch.dict(os.environ, {'INGRESS_DROP_USE_PG_COUNTER': 'true'}, clear=True):
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py]: Style — five nested `with` blocks are hard to scan. Python 3.10+ (PEP 617) supports parenthesized comma-separated context managers:

```python
with (
    patch('probing_base.sai_base_test.ThriftInterfaceDataPlane', MockThriftInterfaceDataPlane),
    patch('probing_base.switch_init') as mock_switch_init,
    patch('probing_base.time.sleep') as mock_sleep,
    patch('probing_observer.ProbingObserver.trace') as mock_trace,
    patch.dict(os.environ, {'INGRESS_DROP_USE_PG_COUNTER': 'true'}, clear=True),
):
    pb.setUp()
```

Defensive aside: `clear=True` is process-wide. Fine for sequential runs (and matches existing convention at line 613), but a theoretical race risk if the suite ever moves to `pytest-xdist`.

File: tests/saitests/mock/ut/test_probing_base.py [hunk 4]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
...
+                            pb.setUp()
+
+        assert pb.use_pg_drop_counter is True
+        assert pb.POINT_PROBING_STEP_SIZE == 1
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py]: This assertion always passes regardless of what `setUp()` does, because:

1. `ConcreteProbingBase.__init__` (line 58) sets `self.POINT_PROBING_STEP_SIZE = 1`.
2. `clear=True` removes the `POINT_PROBING_STEP_SIZE` env var.
3. `setUp()` only overrides the attribute when the env var is set & valid → skips the override.
4. The value remains `1` because of `__init__`, not because `setUp()` preserved anything.

So the assertion is testing `__init__`'s default, not `setUp()` behavior — and that's already covered by `test_setUp_point_probing_step_size_from_env` (order 933) and `_invalid` (order 934).

Either drop the assertion (it's off-topic for the PG-counter test), or add a comment so the intent is explicit:

```python
# POINT_PROBING_STEP_SIZE env unset → setUp() should NOT touch the attribute;
# value of 1 comes from ConcreteProbingBase.__init__, not from setUp().
assert pb.POINT_PROBING_STEP_SIZE == 1
```

File: tests/saitests/mock/ut/test_probing_base.py [hunk 5]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
...
+        mock_switch_init.assert_called_once_with(pb.clients)
+        mock_sleep.assert_called_once_with(5)
+        mock_trace.assert_called()
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py]: Loose assertion — `setUp()` calls `ProbingObserver.trace(...)` 3 times (twice for `POINT_PROBING_STEP_SIZE` before/after env check, once for `use_pg_drop_counter`). `assert_called()` is satisfied as long as *any* of them happened, so a regression that drops the final `use_pg_drop_counter=...` trace wouldn't be caught here.

Tighter version:

```python
trace_messages = [str(c) for c in mock_trace.call_args_list]
assert any("use_pg_drop_counter=True" in m for m in trace_messages), \
    f"expected use_pg_drop_counter=True trace; got {trace_messages}"
```

**Review by XuChen-MSFT — COMMENTED** [review]
Thanks for closing the real-`setUp()` coverage gap that #22545's tests left behind. The mocking strategy is sound — in particular, patching `probing_observer.ProbingObserver.trace` (rather than `probing_base.ProbingObserver.trace`) is exactly the right call for the lazy `from probing_observer import ProbingObserver` inside `setUp()`.

LGTM in principle — all of my comments below are non-blocking polish. None of them block merge; pick whichever ones you find useful.

Quick summary of the inline notes:

1. `pytest.mark.order(9372)` looks like a typo — it breaks the file's strict 910 → 940 incrementing convention and parks this test at the very end of the file's order. Sister tests in `TestProbingBaseSetUp` are 933-937; `9375` (or `941`) would fit naturally.
2. `assert pb.POINT_PROBING_STEP_SIZE == 1` actually verifies `ConcreteProbingBase.__init__`'s default, not anything `setUp()` does — with `clear=True` the env var is unset, so `setUp()` skips its override branch.
3. No baseline guard before `setUp()` runs: if anyone later adds `self.use_pg_drop_counter = True` to `__init__` (even as documentation), the test could become a silent false positive.
4. `mock_trace.assert_called()` is satisfied by any of the 3 trace calls in `setUp()`; it doesn't specifically verify the `use_pg_drop_counter=...` trace.
5. (Style) The 5-level nested `with` could be flattened with PEP 617 parenthesized syntax for readability.
6. (Defensive) `clear=True` is process-wide; theoretical race risk if the suite ever moves to `pytest-xdist` (not enabled today, and matches existing convention at line 613, so no regression here).

#1 and #2 feel worth fixing before merge; #3-6 are author's call. Either way, the test correctly exercises the real `setUp()` path.
